crates/matrix-sdk/src/client/mod.rs

```diff
@@ -1466,26 +1463,11 @@ impl Client {
             versions.push(MatrixVersion::V1_0);
         }
 
-        let unstable_features = resp.unstable_features;
-
-        {
-            // Attempt to cache the result.
-            let encoded = ServerCapabilities::new(&versions, unstable_features.clone());
-            if let Err(err) = self
-                .store()
-                .set_kv_data(
-                    StateStoreDataKey::ServerCapabilities,
-                    StateStoreDataValue::ServerCapabilities(encoded),
-                )
-                .await
-            {
-                warn!("error when caching server capabilities: {err}");
-            }
-        }
-
-        Ok((versions.into(), unstable_features))
+        Ok((versions.into(), resp.unstable_features))
```

Comment:
Might be nicer to return the `ServerCapabilities` struct.

Reply:
Hmm, it makes things a bit awkward:

- if we return `ServerCapabilities` (which holds the stringified Matrix versions), then we need to immediately deserialize it, and handle a potential error that never happens,
- if we return `ClientServerCapabilities` (which has both fields be `Option`al), then we need to immediately unwrap those after calling this method.

I'll keep the current form, as a result, unless you have an idea how to handle this better.